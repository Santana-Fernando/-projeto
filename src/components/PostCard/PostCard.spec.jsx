import { render, screen } from "@testing-library/react"
import { PostCard } from '.'

const mock = {
    id:1,
    title: 'Fernando lindo',
    body:'Body 01',
    cover: 'porno/porno.png'
}

describe('<PostCard/>', () => {
    it('should render PostCard correctly', () => {
        render(<PostCard post={mock} />)

        expect(screen.getByRole('img', {name: /Fernando/i}))
        .toHaveAttribute('src', mock.cover)

        expect(screen.getByRole('heading', {name: /Fernando lindo/i}))
        .toBeInTheDocument()

        expect(screen.getByText('Body 01'))
        .toBeInTheDocument()

    })
    it('should match snapshot', () => {
        const {container} = render(<PostCard post={mock} />)
    
        expect(container.firstChild).toMatchSnapshot()
    })
})