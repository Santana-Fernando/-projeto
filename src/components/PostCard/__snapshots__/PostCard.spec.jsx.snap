// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard/> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Fernando lindo"
    src="porno/porno.png"
  />
  <div
    class="post-content"
  >
    <h1>
      1
    </h1>
    <h2>
      Fernando lindo
    </h2>
    <h3>
      Body 01
    </h3>
  </div>
</div>
`;
